From 84b6dd53f98279cf95769189c660c1ce2d7bb498 Mon Sep 17 00:00:00 2001
From: Jason Wang <jasowang@redhat.com>
Date: Thu, 9 Oct 2014 06:13:20 -0400
Subject: [virt] hyperv/vmbus: Support per-channel driver state

Message-id: <1412835210-22435-2-git-send-email-jasowang@redhat.com>
Patchwork-id: 97283
O-Subject: [RHEL7.1 PATCH V2 01/11] Drivers: hv: vmbus: Support per-channel driver state
Bugzilla: 1146358
RH-Acked-by: Vitaly Kuznetsov <vkuznets@redhat.com>

From: K. Y. Srinivasan <kys@microsoft.com>

Bugzilla: 1146358

As we implement Virtual Receive Side Scaling on the networking side
(the VRSS patches are currently under review), it will be useful to have
per-channel state that vmbus drivers can manage. Add support for
managing per-channel state.

Signed-off-by: K. Y. Srinivasan <kys@microsoft.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
(cherry picked from commit 8a7206a89f85c8439561b0eff85bb9440a2e97a9)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/hyperv.h b/include/linux/hyperv.h
index 4274bdd..4fcd38d 100644
--- a/include/linux/hyperv.h
+++ b/include/linux/hyperv.h
@@ -1071,6 +1071,10 @@ struct vmbus_channel {
   * This will be NULL for the primary channel.
   */
  struct vmbus_channel *primary_channel;
+ /*
+  * Support per-channel state for use by vmbus drivers.
+  */
+ void *per_channel_state;
 };
 
 static inline void set_channel_read_state(struct vmbus_channel *c, bool state)
@@ -1078,6 +1082,16 @@ static inline void set_channel_read_state(struct vmbus_channel *c, bool state)
  c->batched_reading = state;
 }
 
+static inline void set_per_channel_state(struct vmbus_channel *c, void *s)
+{
+ c->per_channel_state = s;
+}
+
+static inline void *get_per_channel_state(struct vmbus_channel *c)
+{
+ return c->per_channel_state;
+}
+
 void vmbus_onmessage(void *context);
 
 int vmbus_request_offers(void);
-- 
1.7.1